src/workloads/query/CollectionLevelDiagnosticCommands.yml

```diff
@@ -0,0 +1,176 @@
+          obj: {int: *integer, string: *string}
+
+# Phase 2: Ensure all data is synced to disk.
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+- Name: DiagnosticTopCommand
+  Type: RunCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3, 5, 7, 9]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 100
+        Thread: 1
+        Database: admin
+        Operations:
+        - OperationMetricsName: DiagnosticTopCommand
+          OperationName: AdminCommand
+          OperationCommand:
+            top: 1
+
+- Name: AggForDiagnosticSetup
+  Type: CrudActor
+  Database: *Database
+  CollectionCount: *CollectionCount
+  Threads: *CollectionCount
+  Phases:
+    OnlyActiveInPhases:
+      Active: [4]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        CollectionCount: *CollectionCount
+        Operations:
+        - OperationMetricsName: AggForDiagnosticSetup
+          OperationName: aggregate
+          OperationCommand:
+            Pipeline: [{$group: {_id: "$int", sumOfDoubles: {$sum: "$double"}}}, {$project: {string: 1}}, {$match: { string : "swimswimswimswimswim" }}]
+
+- Name: QueryForDiagnosticSetup
+  Type: CrudActor
+  Database: *Database
+  CollectionCount: *CollectionCount
+  Threads: *CollectionCount
+  Phases:
+    OnlyActiveInPhases:
+      Active: [6]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 10
+        Database: *Database
+        CollectionCount: *CollectionCount
+        Operation:
+          OperationName: find
+          OperationCommand:
+            Filter: {string: "sinksinksinksink"}
+            Options:
+              Limit: 100
+
+- Name: UpdateManyForDiagnosticSetup
+  Type: CrudActor
+  Database: *Database
+  CollectionCount: *CollectionCount
+  Threads: *CollectionCount
+  Phases:
+    OnlyActiveInPhases:
+      Active: [8]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 5
+        Database: *Database
+        CollectionCount: *CollectionCount
+        Operation:
+          OperationName: updateMany
+          OperationCommand:
+            Filter: { obj.int: {$lt: 5} }
+            Update: {$inc: {x: 1}}
+
+- Name: DeleteManyForDiagnosticSetup
+  Type: CrudActor
+  Database: *Database
+  CollectionCount: *CollectionCount
+  Threads: *CollectionCount
+  Phases:
+    OnlyActiveInPhases:
+      Active: [8]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 5
+        Database: *Database
+        CollectionCount: *CollectionCount
+        Operation:
+          OperationName: deleteMany
+          OperationCommand:
+            Filter: { int: {$eq: 5} }
+
+AutoRun:
+- When:
+    mongodb_setup:
+      $eq:
+      - standalone
```

Comment:
Do you plan to test sharded separately?

Reply:
Yes, because currently top is not supported on a sharded cluster.